CARDS-1506: As a patient, I can review my survey answers before finalizing my submission

acrowthe opened the pull request:
- Add a review page between finishing the last form and being displayed the interpretation screen
  - Has "Update answers" buttons for each form so patients can edit them
- Add a "surveys_submitted" field to the visit information form to track if patients have selected to submit their forms after review
- Minor UI updates (spacing) to interpretation form and summary sections

acrowthe committed:
CARDS-1506: As a patient, I can review my survey answers before finalizing my submission
- Add a review page between finishing the last form and being displayed the
  interpretation screen
  - Has "Update answers" buttons for each form so patients can edit them
- Add a "surveys_submitted" field to the visit information form to track if
  patients have selected to submit their forms after review
- Minor UI updates (spacing) to interpretation form and summary sections

sashaandjic commented:
Update answers button should be more prominent. 
Should we change button label to e.g. `Update AUDIT-C`, `Update PHQ-9`? 
How about dark blue color? 
Move it to the right bottom corner?
Also, notice how AUDIT-C survey is missing from the final screen.

https://user-images.githubusercontent.com/23243244/149605540-8a47c551-edfe-42da-8c74-c1a804e57215.mp4
![image](https://user-images.githubusercontent.com/23243244/149605916-9b600974-bf6d-41a5-97be-c17bbc81c59d.png)



.

sashaandjic commented:
In the Visit information form I would move `Surveys`, "The patient completed the pre-appointment surveys` and `The patient submitted the pre-appointment surveys` fields right below `Time` field, so on the Patient chart they will be shown.
![image](https://user-images.githubusercontent.com/23243244/149606141-57d5d3d9-e995-4990-81cb-b8798eedc9cf.png)
![image](https://user-images.githubusercontent.com/23243244/149606145-90b879c3-8364-4b43-aedd-18207b2a0340.png)

marta- commented:
> In the Visit information form I would move `Surveys`, `The patient completed the pre-appointment surveys` and `The patient submitted the pre-appointment surveys` fields right below `Time` field, so on the Patient chart they will be shown.

Let's move "surveys", "completed" and "submitted" right above "time", this way the fields are somehow sorted chronologically (they get the surveys, they complete, they submit, then it's time for their appointment). Let's also make "surveys" hidden, that's a very technical piece of information.

marta- commented:
> `UPDATE ANSWERS` button should be more prominent. Should we change button label to e.g. `UPDATE AUDIT-C`, `UPDATE PHQ-9`, ...? How about dark blue color? Move it to the right bottom corner?

My fault, I wrote color `primary` instead of `secondary` in the specs. We'll make them `secondary` (red-orange) to start. We should not use the short name of the questionnaire in the patient UI, it might be confusing for them.

acrowthe commented:
Secondary, outlined:
![image](https://user-images.githubusercontent.com/20056751/149794177-6e337263-e760-461b-89e8-3735bc7ac3f2.png)

AUDIT-C missing from interpretation screen reproduced, debugging now.

Visit information form reordered with all fields filled in:
![image](https://user-images.githubusercontent.com/20056751/149794429-c26bc78f-8884-4965-96c6-6a33d9364743.png)

marta- commented:
> Secondary, outlined: 

It needs a bit of space to breathe. Please increase the space between the button and the next survey's title (4 or 5 spacing units might work).

acrowthe committed:
CARDS-1506: As a patient, I can review my survey answers before finalizing my submission
- Improve "Update answers" button styling
- Fix AUDIT-C formula error preventing it's interpretation from being shown
- Reorder Visit information form for better subject view summary

acrowthe commented:
All comments should be addressed and ready for review
- Button style and spacing improved
- AUDIT-C fixed
- Visit information form reordered (Surveys, now hidden, then completed, summary time and the rest)

marta- commented on proms-resources/frontend/src/main/frontend/src/QuestionnaireSet.jsx:
```diff
@@ -321,24 +393,51 @@ function QuestionnaireSet(props) {
+            color="secondary"
+            className={classes.updateButton}
+            onClick={() => {setReviewMode(true); setCrtFormId(subjectData[q]["@name"]); setCrtStep(i)}}>
+              Update answers
```
```suggestion
              Update this survey
```

marta- commented:
The functionality seems to be all good. Any non-blocking UI issues should be listed in and resolved as part of CARDS-1574.

sashaandjic reviewed (approved):
tested

acrowthe committed:
CARDS-1506: As a patient, I can review my survey answers before finalizing my submission
- Clean up disableFooter
- Reword "update answers" button

acrowthe commented:
disableFooter property cleaned up and "update this survey" button text updated, ready for final review/test

sashaandjic reviewed (approved):
re-tested

sdumitriu commented on proms-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Visit information.xml:
```diff
@@ -38,78 +38,97 @@
+		</property>
+		<property>
+			<name>displayMode</name>
+			<value>hidden</value>
```
Why is this hidden?

marta- replied:
Because it's a technical piece of information used only by the service that generates the forms for the patient, and that has no meaning for the clinician.

sdumitriu replied:
But the FHIR ID is even more technical, and we're not hiding that one...
